core/src/main/java/com/linecorp/armeria/client/proxy/WrappingProxyConfigSelector.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.proxy;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Proxy;
+import java.net.ProxySelector;
+import java.net.SocketAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.SessionProtocol;
+
+/**
+ * See {@link ProxyConfigSelector#of(ProxySelector)} for more information.
+ */
+final class WrappingProxyConfigSelector implements ProxyConfigSelector {
+    private static final Logger logger = LoggerFactory.getLogger(WrappingProxyConfigSelector.class);
+
+    /**
+     * Converts {@link SessionProtocol}, {@link Endpoint} to a uri without serialization format.
+     */
+    private static URI toUri(SessionProtocol sessionProtocol, Endpoint endpoint) {
+        try {
```

Comment:
I feel like this change deserves some explanation:

`Endpoint.toUri` requires a serialization format (which was always `none` before introducing this method)
However, serialization format may not necessarily be `none`. (i.e. `tbinary` calls with proxy)
On the other hand, a `session` shouldn't necessarily only use a single serialization format, hence I moved away from using `Scheme` for the callback methods.
Hence, I'd decided to omit `SerializationFormat` entirely from the uri scheme.

Reply:
OK!